sycl/include/sycl/builtins.hpp

```diff
@@ -170,6 +187,96 @@ inline __SYCL_ALWAYS_INLINE
+      res[N - 1] = __sycl_std::__invoke_##NAME<T>(x[N - 1], y_vec[0]);         \
+    }                                                                          \
+    return res;                                                                \
+  }
+
+__SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD(fmax)
+__SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD(fmin)
+
+#undef __SYCL_MATH_FUNCTION_2_SGENFLOAT_Y_OVERLOAD
+
+        template <typename T, size_t N>
+        inline __SYCL_ALWAYS_INLINE
+    std::enable_if_t<detail::is_sgenfloat<T>::value, marray<T, N>> ldexp(
+        marray<T, N> x, marray<int, N> k) __NOEXC {
+  marray<T, N> res;
+  for (size_t i = 0; i < N; i++) {
+    res[i] = __sycl_std::__invoke_ldexp<T>(x[i], k[i]);
+  }
+  return res;
+}
+
+template <typename T, size_t N>
+inline __SYCL_ALWAYS_INLINE
+    std::enable_if_t<detail::is_sgenfloat<T>::value, marray<T, N>>
+    ldexp(marray<T, N> x, int k) __NOEXC {
+  marray<T, N> res;
+  for (size_t i = 0; i < N; i++) {
+    res[i] = __sycl_std::__invoke_ldexp<T>(x[i], k);
+  }
+  return res;
+}
+
+#define __SYCL_MATH_FUNCTION_2_GENINT_Y_OVERLOAD_IMPL(NAME)                    \
+  marray<T, N> res;                                                            \
+  for (size_t i = 0; i < N; i++) {                                             \
+    res[i] = __sycl_std::__invoke_##NAME<T>(x[i], y[i]);                       \
+  }                                                                            \
+  return res;
+
+template <typename T, size_t N>
+inline __SYCL_ALWAYS_INLINE
+    std::enable_if_t<detail::is_sgenfloat<T>::value, marray<T, N>>
+    pown(marray<T, N> x, marray<int, N> y) __NOEXC {
+  __SYCL_MATH_FUNCTION_2_GENINT_Y_OVERLOAD_IMPL(pown)
+}
+
+template <typename T, size_t N>
+inline __SYCL_ALWAYS_INLINE
+    std::enable_if_t<detail::is_sgenfloat<T>::value, marray<T, N>>
+    rootn(marray<T, N> x, marray<int, N> y) __NOEXC {
+  __SYCL_MATH_FUNCTION_2_GENINT_Y_OVERLOAD_IMPL(rootn)
+}
+
+#undef __SYCL_MATH_FUNCTION_2_GENINT_Y_OVERLOAD_IMPL
+
+#define __SYCL_MATH_FUNCTION_2_INT_Y_OVERLOAD_IMPL(NAME)                       \
+  marray<T, N> res;                                                            \
+  for (size_t i = 0; i < N; i++) {                                             \
+    res[i] = __sycl_std::__invoke_##NAME<T>(x[i], y);                          \
+  }                                                                            \
+  return res;
+
+template <typename T, size_t N>
+inline __SYCL_ALWAYS_INLINE
+    std::enable_if_t<detail::is_sgenfloat<T>::value, marray<T, N>>
+    pown(marray<T, N> x, int y) __NOEXC {
+  __SYCL_MATH_FUNCTION_2_INT_Y_OVERLOAD_IMPL(pown)
+}
+
+template <typename T, size_t N>
+inline __SYCL_ALWAYS_INLINE
+    std::enable_if_t<detail::is_sgenfloat<T>::value, marray<T, N>>
+    rootn(marray<T, N> x,
+          int y) __NOEXC{__SYCL_MATH_FUNCTION_2_INT_Y_OVERLOAD_IMPL(rootn)}
```

Comment:
```suggestion
std::enable_if_t<detail::is_sgenfloat<T>::value, marray<T, N>>
    rootn(marray<T, N> x, int y) __NOEXC {
  __SYCL_MATH_FUNCTION_2_INT_Y_OVERLOAD_IMPL(rootn)
}
```

Looks like clang-format got confused by all those macro?

Reply:
https://github.com/intel/llvm/pull/8631/commits/23cfc1e139caef4be3391dd9eaccf66d4c7d39d4